### ISSUE TITLE
Homepage slow frame rate = 100% 

### ISSUE THREAD

**Issue by mafeng-Bryant** [getsentry/sentry-cocoa#1881]
### Platform

iOS

### Installed

CocoaPods

### Version

7.15.0

### Steps to Reproduce

1. We need to count the frame rate of the page and integrate the sentry sdk, but we see that the data panel shows that the slow frame rate of our home page is 100%, why?

### Expected Result

We hope that the slow frame rate of the home page is 4.50%, and the statistics of other platforms are about 4.50%.

### Actual Result

Homepage slow frame rate = 100%

**Comment by philipphofmann** [comment]
I'm not sure why the slow frame rate of your home page is 100%. Can you please give us more context on how we could reproduce the issue? How does your app setup look? Does this happen every time? On which devices does this happen?

**Comment by mafeng-Bryant** [comment]
> I'm not sure why the slow frame rate of your home page is 100%. Can you please give us more context on how we could reproduce the issue? How does your app setup look? Does this happen every time? On which devices does this happen?

I initialize the sentry SDK in the willfinishlaunchingwithoptions method. The initialization code is as follows：
[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
options.dsn = kSentryProductKey;
options.tracesSampleRate = @(1.0);
options.debug = NO;
options.releaseName = kAppVersion;
options.enableOutOfMemoryTracking = NO;
[options setBeforeSend:^SentryEvent * _Nullable(SentryEvent * _Nonnull event) {
if (event.exceptions && event.exceptions.count >0) {
id type = [event.exceptions lastObject];
if (ISCLASS([SentryException class], type)) {
return NULL;
}
}
return event;
}];
}];
device info: on all iPhone device

**Comment by mafeng-Bryant** [comment]
> > the statistics of other platforms are about 4.50%.
> 
> Which statistics of which platforms?
> 
> What exactly do you mean by home page? Is the slow frame rate 100% for every screen on your app or just one screen?

platform is firebase 

just home screen

**Comment by philipphofmann** [comment]
Are you maybe testing this on an iPhone 13 Pro with 120 fps? Firebase uses a similar approach to calculate the frame rate, but they always assume 60 fps, see 
```
// Note: This was previously 60 FPS, but that resulted in 90% +  of all frames collected to be
// flagged as slow frames, and so the threshold for iOS is being changed to 59 FPS.
// TODO(b/73498642): Make these configurable.
CFTimeInterval const kFPRSlowFrameThreshold = 1.0 / 59.0;  // Anything less than 59 FPS is slow.
```
https://github.com/firebase/firebase-ios-sdk/blob/e8828e8efc813817b3c2cbfde8e965f7a7a26641/FirebasePerformance/Sources/AppActivity/FPRScreenTraceTracker.m#L28-L31

**Comment by brustolin** [comment]
Because devices with ProMotion, we need to update the frame rate calculation.
Documentation suggest:

```objc
// Calculate the actual frame rate.
double actualFramesPerSecond = 1 / (displaylink.targetTimestamp - displaylink.timestamp);
```

**Comment by mafeng-Bryant** [comment]
> I don't know if this is the problem here, but because devices with ProMotion, we need to update the frame rate calculation. Documentation suggest:
> 
> ```objc
> // Calculate the actual frame rate.
> double actualFramesPerSecond = 1 / (displaylink.targetTimestamp - displaylink.timestamp);
> ```

Which version will you update? I am using IOS sentry SDK version = 7.15.0

**Comment by mafeng-Bryant** [comment]
> Are you maybe testing this on an iPhone 13 Pro with 120 fps? Firebase uses a similar approach to calculate the frame rate, but they always assume 60 fps, see
> 
> ```
> // Note: This was previously 60 FPS, but that resulted in 90% +  of all frames collected to be
> // flagged as slow frames, and so the threshold for iOS is being changed to 59 FPS.
> // TODO(b/73498642): Make these configurable.
> CFTimeInterval const kFPRSlowFrameThreshold = 1.0 / 59.0;  // Anything less than 59 FPS is slow.
> ```
> 
> https://github.com/firebase/firebase-ios-sdk/blob/e8828e8efc813817b3c2cbfde8e965f7a7a26641/FirebasePerformance/Sources/AppActivity/FPRScreenTraceTracker.m#L28-L31
> 
> We get the actual frame rate of the phone
> 
> https://github.com/getsentry/sentry-cocoa/blob/5543879c09e95866abacf24d6dfb4487c0f62cb7/Sources/Sentry/SentryFramesTracker.m#L59
> 
> Maybe this is not working properly for us.



> 

iPhone13,4
iPhone14,3
iPhone11,8
iPhone13,2
iPhone12,5
iPhone14,5
iPhone13,3    这些是我们的设备

**Comment by mafeng-Bryant** [comment]
> Are you maybe testing this on an iPhone 13 Pro with 120 fps? Firebase uses a similar approach to calculate the frame rate, but they always assume 60 fps, see
> 
> ```
> // Note: This was previously 60 FPS, but that resulted in 90% +  of all frames collected to be
> // flagged as slow frames, and so the threshold for iOS is being changed to 59 FPS.
> // TODO(b/73498642): Make these configurable.
> CFTimeInterval const kFPRSlowFrameThreshold = 1.0 / 59.0;  // Anything less than 59 FPS is slow.
> ```
> 
> https://github.com/firebase/firebase-ios-sdk/blob/e8828e8efc813817b3c2cbfde8e965f7a7a26641/FirebasePerformance/Sources/AppActivity/FPRScreenTraceTracker.m#L28-L31
> 
> We get the actual frame rate of the phone
> 
> https://github.com/getsentry/sentry-cocoa/blob/5543879c09e95866abacf24d6dfb4487c0f62cb7/Sources/Sentry/SentryFramesTracker.m#L59
> 
> Maybe this is not working properly for us.

iPhone13,4
iPhone14,3
iPhone11,8
iPhone13,2
iPhone12,5
iPhone14,5
iPhone13,3  
This is our equipment

**Comment by philipphofmann** [comment]
> iPhone13,4
> iPhone14,3
> iPhone11,8
> iPhone13,2
> iPhone12,5
> iPhone14,5
> iPhone13,3
> This is our equipment

I'm a bit confused. What does this stand for? There is no iPhone 14 yet, @mafeng-Bryant.

**Comment by philipphofmann** [comment]
But this means this also happens before iPhone 13 Pro. So the promotion display is not the cause for this issue.

**Comment by brustolin** [comment]
Yes. That's what I thought. But, we still need to fix the frame rate calculation.

@mafeng-Bryant have you changed `CADisplayLink.preferredFramesPerSecond`?

**Comment by ialimz** [comment]
We had this issue too. Any updates?
Many of our transactions has slow frames 100%

**Comment by philipphofmann** [comment]
@ialimz, since when do you see this issue?

**Comment by philipphofmann** [comment]
I just found out from a customer that this is happening at least since April, so no change in the latest version is responsible for this.
